### DISCUSSION TITLE
Home Assistant statistics

### DISCUSSION THREAD

**Discussion by Ivano62** [astrandb/miele#208] [Q&A]
### The problem

The sensor Oven Target temperature (sensor.oven_target_temperature °C) generate this message "The state class '' of this entity is not supported."

It seems that the class is not defined.

### Version of miele integration having the issue?

0.1.4

### Version of Home Assistant Core having the issue?

2023.1.4

**Comment by astrandb** [comment]
This is by design. Settings like target temperature shall not generate long-term statistics according to Home Assistant design principles.

**Comment by Ivano62** [comment]
Okay. Thank you.

However, I don't understand why only for this sensor the problem is reported?
I have a ton of temperature sensors that are routinely run for long term stats as well.

**Comment by astrandb** [comment]
We have chosen to follow the development guidelines rather strictly. The definitions for state_class can be read here https://developers.home-assistant.io/docs/core/entity/sensor#available-state-classes. Thus a target temperature is not a measurement. It is a setting that you set yourself, either on the appliance or in the UI.

If you need long-time statistics on this setting,  you can make a workaround by creating a template sensor and assign what state_class you find appropriate.

**Comment by Ivano62** [comment]
Evidently I explained myself badly, I have no need for long-term statistics for this parameter.
I only signaled the presence of the message in the Home Assistant, which does not happen for other parameters of other sensors.
Thank you
